src/hash.h

```diff
@@ -171,16 +171,18 @@ unsigned int MurmurHash3(unsigned int nHashSeed, const std::vector<unsigned char
 
 void BIP32Hash(const ChainCode &chainCode, unsigned int nChild, unsigned char header, const unsigned char data[32], unsigned char output[64]);
 
-/** SipHash-2-4, using a uint64_t-based (rather than byte-based) interface */
+/** SipHash-2-4 */
 class CSipHasher
 {
 private:
     uint64_t v[4];
+    uint64_t tmp;
     int count;
 
 public:
     CSipHasher(uint64_t k0, uint64_t k1);
     CSipHasher& Write(uint64_t data);
+    CSipHasher& Write(const unsigned char* data, size_t size);
```

Comment:
Should probably note somewhere that the two Write methods are partially-mutually-exclusive per-object.